src/main/java/com/google/sps/api/GmailAPI.java

```diff
@@ -0,0 +1,142 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.api;
+
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.api.client.googleapis.auth.oauth2.GoogleClientSecrets;
+import com.google.api.client.googleapis.auth.oauth2.GoogleCredential;
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.client.util.StringUtils;
+import com.google.api.services.gmail.Gmail;
+import com.google.api.services.gmail.model.ListMessagesResponse;
+import com.google.api.services.gmail.model.Message;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.net.URLEncoder;
+import java.security.GeneralSecurityException;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import org.apache.commons.codec.binary.Base64;
+import org.json.JSONObject;
+
+/** Set up Gmail API credentails. */
+public class GmailAPI {
+
+	private static final String APPLICATION_NAME = "noms";
+	private static final JsonFactory JSON_FACTORY = JacksonFactory.getDefaultInstance();
+	private static final String USER = "me";
+  private static final File CREDENTIALS_FILE_PATH = new File(System.getProperty("user.dir") + "/credentials.json");
+  
+  private static String CLIENT_ID = "";
+  private static String CLIENT_SECRET = "";
+	private static Gmail service = null;
+
+  /**
+    * Create a Gmail service using credentails.
+    *
+    * @throws IOException
+    * @throws GeneralSecurityException
+    */
+	public static Gmail getGmailService() throws IOException, GeneralSecurityException {
+
+    // Gather client info from OAuth 2.0 credentials.
+		InputStream in = new FileInputStream(CREDENTIALS_FILE_PATH); 
+		GoogleClientSecrets clientSecrets = GoogleClientSecrets.load(JSON_FACTORY, new InputStreamReader(in));
+    CLIENT_ID = clientSecrets.getDetails().getClientId().toString();
+    CLIENT_SECRET = clientSecrets.getDetails().getClientSecret().toString();
+
+    // Set up credentials to use the Gmail API.
+		Credential authorize = new GoogleCredential.Builder().setTransport(GoogleNetHttpTransport.newTrustedTransport())
+				.setJsonFactory(JSON_FACTORY)
+				.setClientSecrets(CLIENT_ID, CLIENT_SECRET)
+				.build()
+        .setRefreshToken(Token.REFRESH_TOKEN)
+        .setAccessToken(getAccessToken(Token.REFRESH_TOKEN));
```

Comment:
Since it is possible to return `null` from `getAccessToken`, I would like to see some error handling done. A TODO is fine for now.

Reply:
Gotcha, will do!